hazelcast/src/main/java/com/hazelcast/internal/diagnostics/EventQueuePlugin.java

```diff
@@ -0,0 +1,242 @@
+            clear();
+            scan(writer, eventQueue, index++);
+        }
+
+        writer.endSection();
+    }
+
+    // just for testing
+    ItemCounter<String> getOccurrenceMap() {
+        return occurrenceMap;
+    }
+
+    private List<BlockingQueue<Runnable>> getEventQueues() {
+        return eventExecutor.getWorkQueues();
+    }
+
+    private void scan(DiagnosticsLogWriter writer, BlockingQueue<Runnable> eventQueue, int index) {
+        int sampleCount = sample(eventQueue);
+        if (sampleCount < 0) {
+            return;
+        }
+
+        render(writer, sampleCount, index);
+    }
+
+    private void render(DiagnosticsLogWriter writer, int sampleCount, int index) {
+        writer.startSection("worker=" + index);
+
+        writer.writeKeyValueEntry("eventCount", events.size());
+        writer.writeKeyValueEntry("sampleCount", sampleCount);
+        renderSamples(writer, sampleCount);
+
+        writer.endSection();
+    }
+
+    private void renderSamples(DiagnosticsLogWriter writer, int sampleCount) {
+        writer.startSection("samples");
+
+        for (String key : occurrenceMap.keySet()) {
+            long value = occurrenceMap.get(key);
+            if (value == 0) {
+                continue;
+            }
+
+            double percentage = (1d * value) / sampleCount;
+            writer.writeEntry(key + " sampleCount=" + value + " " + defaultFormat.format(percentage));
+        }
+        writer.endSection();
+    }
+
+    private void clear() {
+        occurrenceMap.reset();
+        events.clear();
+    }
+
+    /**
+     * Samples the queue.
+     *
+     * @param queue the queue to sample
+     * @return the number of samples, or -1 if there were not sufficient samples
+     */
+    private int sample(BlockingQueue<Runnable> queue) {
+        for (Runnable runnable : queue) {
+            events.add(runnable);
+        }
+
+        int eventCount = events.size();
+        if (eventCount < threshold) {
+            return -1;
+        }
+
+        int sampleCount = min(samples, eventCount);
+        int actualSampleCount = 0;
+        while (actualSampleCount < sampleCount) {
+            Runnable runnable = events.get(random.nextInt(eventCount));
+            actualSampleCount += sampleRunnable(runnable);
+        }
+
+        return actualSampleCount;
+    }
+
+    int sampleRunnable(Runnable runnable) {
+        if (runnable instanceof LocalEventDispatcher) {
+            LocalEventDispatcher eventDispatcher = (LocalEventDispatcher) runnable;
+            Object dispatcherEvent = eventDispatcher.getEvent();
+            if (dispatcherEvent instanceof EntryEventData) {
+                // IMap event
+                EntryEventData event = (EntryEventData) dispatcherEvent;
+                EntryEventType type = EntryEventType.getByType(event.getEventType());
+                String mapName = event.getMapName();
+                occurrenceMap.add(format("IMap '%s' %s", mapName, type), 1);
```

Comment:
I accidentally printed the whole events array (because I misread the connection plugin to do that). That was my main concern, because that created a huge String monster. This is not really efficient, but I don't see an easy solution to prevent the key generation for each sample. There doesn't seem to be a better identifier, so we could just do this once in the renderer.